### ISSUE TITLE
Create a way to exclude few repositories.

### ISSUE THREAD

**Issue by suyash-thakur** [suyash-thakur/js-good-first-issues-finder#6]
A list of repositories that needed to be excluded from the list. There are few repositories that are intended to learn git and open-source. Although these repos are actually usefully but I think doesn't belong to this list.

**Describe the solution you'd like**
We can have a json file containing the repo name needed to be excluded and while fetching the repos if we encounter any of these repos we can skip it, instead of adding it to the `README.md`

**Describe alternatives you've considered**
I am open to any alternative solution here. 

**Additional context**
As an example repo that needs to be excluded is -: 
[https://github.com/fork-commit-merge/fork-commit-merge](https://github.com/fork-commit-merge/fork-commit-merge)

**Comment by saksham-jain177** [comment]
How about adding a specific tag to the repos that need be excluded, to filter out repos with that tag

**Comment by suyash-thakur** [comment]
@saksham-jain177  The issue is there is no unique way to identify the repos with tags. Most of the issues in these repos are tagged `good-first-issue`. The only way I can think of is to maintain a manual black list so that we don't end up excluding repos that actually belongs to the list.

I also didn't understand the "adding tags to repos" part. These repos are maintained by different people and we can't add tags to them from our end.

**Comment by Ishpreet01** [comment]
@suyash-thakur Does this mean that are only a fixed set of repositories that need to be excluded? Or else we cannot specify the
repositories in a json file to exclude them while fetching the other repositories

**Comment by suyash-thakur** [comment]
@Ishpreet01  The repository that needed to be excluded is a manual process. There are some repos which are for people to learn basic of git (creating PR and stuff). There is no unique identifier for these repos. A json file for listing these repos will be good enough. We will keep adding the similar kind of repos to the list as we keep encountering them.

**Comment by abishek1740P** [comment]
To achieve the goal of excluding specific repositories while fetching and listing them, you can create a JSON file containing the names of the repositories to be excluded. Then, you can modify your script to skip these repositories when generating the README.md file.

**Comment by suyash-thakur** [comment]
@abishek1740P Yup. That's the only solution that I can think of now. Can you create a PR for it ?

**Comment by edenmengste** [comment]
> [@abishek1740P](https://github.com/abishek1740P) Yup. That's the only solution that I can think of now. Can you create a PR for it ?
I can try. Does it accept user input? and since it is beginner friendly does the code only generates the JSON data within the browser?

**Comment by suyash-thakur** [comment]
@edenmengste It should be a static list of repos. Shouldn't take any user input. The JSON should be a constant list that script checks while fetching the issues using the Github API. This should filter out some repos in the creation process.

**Comment by Rocklinks** [comment]
This can be achieved by creating a simple excluded_repos.json file that contains the list of repository names (or URLs) to ignore. When generating or updating the README.md, the script should check against this list and skip any repo that matches.

This way, we keep the logic and data separate ,allowing for easier maintenance and future additions to the exclusion list.

**Comment by suyash-thakur** [comment]
@Rocklinks  Yup that's the best solution. I was also wondering if the API itself takes an exclusion parameter but will have to check that out.

**Comment by PrathameshWankhede4** [comment]
I suggest we add an exclude.json file to the root of the repo that contains a list of repository names to exclude, like:

json
Copy
Edit
[
  "fork-commit-merge/fork-commit-merge",
  "username/test-repo"
]
While generating the README.md, we can read this file and skip any repos listed in it. This way, we can keep the logic modular and easily update the list without changing the script every time. Let me know if you'd like me to raise a PR for this!

**Comment by niveditasargar** [comment]
Hi 👋, I would like to work on this issue.
My plan is to add an exclude.json file listing repositories to be excluded.
While fetching repositories, the script will skip any listed in exclude.json, so they won’t be added to the README.md.
Could you please assign this issue to me?

**Comment by Amirthavarshini04** [comment]
Hi @suyash-thakur 

I'm Amirthavarshini, a final-year Computer Science student and I'm looking to make my first open source contribution — this issue looks like a great fit!

I've gone through the entire discussion thread and the solution is clear:
- Create an `excluded_repos.json` file in the root with a list of repo names to exclude
- Modify `index.js` to read this file and skip matching repos while fetching issues from the GitHub API

I'm comfortable with JavaScript and JSON, and I can get started immediately. Could you please assign this issue to me?

Thank you for maintaining this project!

**Comment by Vedant43hh** [comment]
Hi! I would like to work on this issue. Could I please be assigned to it?
